### PR TITLE
US-9-Homepagina aanbod

### PR DESCRIPTION
**Pull request by Alyvox**

### CODE REVIEW

**Comment by Alyvox** [comment]
[Homepagina aanbod](https://trello.com/c/GH1fhIt2/9-homepagina-aanbod)

**Comment by Alyvox** [comment]
Change list:
- Added unit tests for traileroffers GET API
- Added pagination for traileroffers GET API
- Made traileroffers API restful

File: api/src/main/java/com/buurbak/api/trailers/dto/TrailerInfoDTO.java [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+import lombok.Data;
 
 import javax.validation.constraints.NotBlank;
 import java.util.UUID;
 
-public record TrailerInfoDTO(@NotBlank(message = "Id may not be blank") UUID id,
-                             @NotBlank(message = "TrailerType may not be blank") @JsonAlias("trailer_type") TrailerType trailerType,
-                             @NotBlank(message = "Location may not be blank") String location,
-                             @NotBlank(message = "Price may not be blank") double price) {
+@Data
```
Comment (spark-156) [inline api/src/main/java/com/buurbak/api/trailers/dto/TrailerInfoDTO.java:8]:
```suggestion
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
```

File: api/src/main/java/com/buurbak/api/trailers/dto/TrailerTypeDTO.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+@Getter
+@Setter
+@NoArgsConstructor
+public class TrailerTypeDTO {
+    @NotBlank(message = "Name may not be blank")
+    @JsonAlias("trailer_type")
+    private String name;
+
+    public TrailerTypeDTO(String name){this.name = name;}
```
Comment (spark-156) [inline api/src/main/java/com/buurbak/api/trailers/dto/TrailerTypeDTO.java:18]:
```suggestion
```

File: api/src/main/java/com/buurbak/api/trailers/dto/TrailerTypeDTO.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import com.fasterxml.jackson.annotation.JsonAlias;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.validation.constraints.NotBlank;
+
+@Getter
+@Setter
+@NoArgsConstructor
```
Comment (spark-156) [inline api/src/main/java/com/buurbak/api/trailers/dto/TrailerTypeDTO.java:12]:
```suggestion
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
```

**Review by spark-156 — CHANGES_REQUESTED** [review]

**Pull request merged**